src/interface.jl

```diff
@@ -81,3 +81,73 @@ to checking if the value returned by `param_sym_to_index` is not `nothing`.
 function is_param_sym end
 
 is_param_sym(store, sym) = !isnothing(param_sym_to_index(store, sym))
+
+"""
+$(TYPEDSIGNATURES)
+
+Get an iterable over the observed variable expressions for the given system.
+Default to an empty vector.
+"""
+function observed end
+
+observed(::Any) = []
+
+"""
+$(TYPEDSIGNATURES)
+
+Check if the given sym is an observed variable in the given system. Default
+to checking if the value returned by `observed_sym_to_index` is not `nothing`.
+"""
+function is_observed_sym end
+
+is_observed_sym(store, sym) = !isnothing(observed_sym_to_index(store, sym))
+
+"""
+$(TYPEDSIGNATURES)
+
+Find the index of the given sym in the given system. Default to the index of the first
+symbol in the iterable returned by `states` which matches the given `sym`. Return
+`nothing` if the given `sym` does not match.
+"""
+function observed_sym_to_index end
+
+function observed_sym_to_index(store, sym)
+    findfirst(isequal(Symbol(sym)), Symbol.(map(o -> o.lhs, observed(store))))
```

Comment:
This allocates quite a bit, but it doesn't have to. Also, we don't need to convert `sym` to a `Symbol` to compare.